packages/discord-player/src/queue/GuildQueuePlayerNode.ts

```diff
@@ -686,7 +686,8 @@ export class GuildQueuePlayerNode<Meta = unknown> {
         this.queue.emit(GuildQueueEvent.playerSkip, this.queue, track, TrackSkipReason.NoStream, streamDefinitelyFailedMyDearT_TPleaseTrustMeItsNotMyFault.message);
         this.queue.emit(GuildQueueEvent.playerError, this.queue, streamDefinitelyFailedMyDearT_TPleaseTrustMeItsNotMyFault, track);
         const nextTrack = this.queue.tracks.dispatch();
-        if (nextTrack) this.play(nextTrack, { queue: false });
+        if (nextTrack) return void this.play(nextTrack, { queue: false });
+        this.queue.dispatcher.emit("finish");
         return;
```

Comment:
this can be omitted

Reply:
but its not triggering the emptyQueue event without it?

Reply:
no, i mean the last `return;` line, it has no purpose

Reply:
oh got you